manifests/init.pp

```diff
@@ -116,6 +116,12 @@
   $wait_for_pxp_agent_exit = undef,
 ) inherits ::puppet_agent::params {
 
+  if any_resources_of_type('filebucket', { path => false }) {
+    if $settings::digest_algorithm != $::puppet_digest_algorithm {
```

Comment:
I guess this condition will pass if agent nodes are upgraded prior to server nodes, so then we'd get to sha256 on agent and md5 on server which is still incompatible.

We also wouldn't know whether `digest_algorithm` is explicitly set in `puppet.conf` or the default value... which raises more questions.

Reply:
[Official documentation](https://puppet.com/docs/puppet/latest/upgrade_minor.html) states that server nodes should always be upgraded first, so my previous concern is unfounded.

Reply:
Reference for the `$settings::{setting}` variables: https://puppet.com/docs/puppet/5.5/lang_facts_and_builtin_vars.html#puppet-master-variables